### DISCUSSION TITLE
Fixing dates in an xr dataset

### DISCUSSION THREAD

**Discussion by kholl03** [giswqs/geog-312#100] [Q&A] [answered]
Hi all, 

I'm trying to select a certain day in my xr dataset. For reference, I'm using the 'rasm' tutorial dataset xr supplies.

When I opened this file, I had to manually decode times because simply loading in the dataset wouldn't do it. I used this code:

`ds['time'] = cftime.num2date(ds['time'].values, ds['time'].attrs['units'])`

I think this messed up the dates, as I think the actual dataset is meant to start on 1979-09-01, but after decoding, mine starts at 1979-05-23. Is there a way I can fix this? 

I need to fix this because I need to find data for 1980-07-01.

`selected_tair = air_temp.sel(time='1980-03-22', y=70, method = 'nearest')` 

(Here I'm using 1980-03-22 because I think that's the real date I need after looking at the arrays)

Using that code I get this error: 

`ValueError: cannot supply selection options {'method': 'nearest', 'tolerance': None} for dimension 'y'that has no associated coordinate or index`

I'm not really sure what to do here. I've messed up so bad that I can't even comprehend it.

**Comment by giswqs** [comment, marked as answer]
Lab 6 has been updated. Please re-download it. 
https://geog-312.gishub.org/book/labs/lab_06.html
